tests/scene/test_image_texture.h

```diff
@@ -0,0 +1,119 @@
+/**************************************************************************/
+/*  test_image_texture.h                                                  */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_IMAGE_TEXTURE_H
+#define TEST_IMAGE_TEXTURE_H
+
+#include "core/io/image.h"
+#include "scene/resources/image_texture.h"
+
+#include "tests/test_macros.h"
+#include "tests/test_utils.h"
+
+namespace TestImageTexture {
+
+TEST_CASE("[SceneTree][ImageTexture][ImageTexture2D] constructor") {
```

Comment:
It has nothing to do with SceneTree, also not sure why the ImageTexture2D subcategory.
```suggestion
TEST_CASE("[ImageTexture] constructor") {
```
(same for other test cases)

Reply:
Thanks for the feedback

Regarding [SceneTree], several `ImageTexture` functions access the `RenderServer`. Adding `[SceneTree]` to the name of tests appears to conditionally initialize the render server. If it's not in the name, then these functions segfault on these access calls. It does seem hacky, but many other unit tests also have this (92 results for "[SceneTree]" in the tests folder across 23 files).

Regarding the [ImageTexture2D], I added that with the intention to leave room for unit testing the other 5 classes in the `image_texture.h` file in this file if someone wanted to later. ex: 
`TEST_CASE("[SceneTree][ImageTexture][ImageTexture3D] constructor") {...}`
`TEST_CASE("[SceneTree][ImageTexture][ImageTextureLayered] constructor") {...}`
I'm fine removing that though.

Reply:
> Regarding `[SceneTree]`, several `ImageTexture` functions access the `RenderServer`. Adding `[SceneTree]` to the name of tests appears to conditionally initialize the render server (test_main.cpp:230). If it's not in the name, then these functions segfault on these access calls. It does seem hacky, but many other unit tests also have this (92 results for "[SceneTree]" in the tests folder across 23 files).

Oh, I wasn't aware of that. I'd say a single short comment (before the first test case?) justifying the usage of [SceneTree] might be a good idea for future me-alikes. :wink:



> Regarding the `[ImageTexture2D]`, I added that with the intention to leave room for unit testing the other 5 classes in the `image_texture.h` file in this file if someone wanted to later. ex:
> `TEST_CASE("[SceneTree][ImageTexture][ImageTexture3D] constructor") {...}`
> `TEST_CASE("[SceneTree][ImageTexture][ImageTextureLayered] constructor") {...}`
> I'm fine removing that though.

I'm not that familiar with the tests so not sure what's preferred here, hence [TIWAGOS](https://www.urbandictionary.com/define.php?term=TIWAGOS). But if we want to keep it something like `[file][class]` then it should rather be `[ImageTexture][ImageTexture]`, as the class being tested is `ImageTexture`, not `ImageTexture2D`. For me the file part makes little sense though so I'd stick with just the class like: `[SceneTree][ImageTexture]`, `[SceneTree][ImageTexture3D]`, etc.

Reply:
That's all fair, I've cleaned up the test case names and added the comment.